tests/spacy/test_spacy_model_export.py

```diff
@@ -0,0 +1,220 @@
+import os
+import random
+from collections import namedtuple
+
+import pandas as pd
+import pytest
+import six
+import spacy
+import yaml
+from spacy.util import compounding, minibatch
+
+import mlflow.spacy
+from sklearn.datasets import fetch_20newsgroups
+
+from mlflow import pyfunc
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils.model_utils import _get_flavor_configuration
+
+ModelWithData = namedtuple("ModelWithData", ["model", "inference_data"])
+
+
+@pytest.fixture
+def spacy_model_with_data():
+    # Creating blank model and setting up the SpaCy pipeline
+    nlp = spacy.blank("en")
+    textcat = nlp.create_pipe(
+        "textcat", config={"exclusive_classes": True, "architecture": "simple_cnn"}
+    )
+    nlp.add_pipe(textcat, last=True)
+
+    # Training the model to recognize between computer graphics and baseball in 20newsgroups dataset
+    categories = ['comp.graphics', 'rec.sport.baseball']
+    for cat in categories:
+        textcat.add_label(cat)
+
+    # Split train/test and train the model
+    train_x, train_y, test_x, _ = _get_train_test_dataset(categories)
+    train_data = list(zip(train_x, [{"cats": cats} for cats in train_y]))
+    _train_model(nlp, train_data)
+
+    return ModelWithData(nlp, pd.DataFrame(test_x))
+
+
+@pytest.fixture
+def spacy_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(
+        conda_env,
+        additional_conda_deps=["pytest"],
+        additional_pip_deps=["spacy"])
+    return conda_env
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+@pytest.mark.large
+def test_model_save_load(spacy_model_with_data, model_path):
+    spacy_model = spacy_model_with_data.model
+    mlflow.spacy.save_model(spacy_model=spacy_model, path=model_path)
+    loaded_model = mlflow.spacy.load_model(model_path)
+
+    # Comparing the meta dictionaries for the original and loaded models
+    assert spacy_model.meta == loaded_model.meta
+
+    # Load pyfunc model using saved model and asserting its predictions are equal to the created one
+    pyfunc_loaded = mlflow.pyfunc.load_pyfunc(model_path)
+    assert all(_predict(spacy_model, spacy_model_with_data.inference_data) ==
+               pyfunc_loaded.predict(spacy_model_with_data.inference_data))
+
+
+@pytest.mark.large
+def test_model_log(spacy_model_with_data):
+    spacy_model = spacy_model_with_data.model
+    old_uri = mlflow.get_tracking_uri()
+    # should_start_run tests whether or not calling log_model() automatically starts a run.
+    for should_start_run in [False, True]:
+        with TempDir(chdr=True, remove_on_exit=True):
+            try:
+                artifact_path = "model"
+                mlflow.set_tracking_uri("test")
```

Comment:
You should be able to use the `tracking_uri_mock` fixture from tests.projects.utils to avoid having to explicitly mock out the tracking URI here, see e.g. https://github.com/mlflow/mlflow/blob/360c0cada0174b71500a83ded72260b6b6942d90/tests/sklearn/test_sklearn_model_export.py#L159

Reply:
Thanks for the the tip. I have implemented it in the latest commit.

Reply:
For some reason travis CI is giving me the following error: 

tests/spacy/test_spacy_model_export.py (91,42): [W0613 unused-argument] Unused argument 'tracking_uri_mock'

even tho I explicitly disabled the warning for pylint:

https://github.com/mlflow/mlflow/blob/dcc0b0930daf4661d68a89761fc619dfa5cd3a30/tests/spacy/test_spacy_model_export.py#L21

Any ideas?